weave/cookbooks/source/Intro_to_Weave_Hello_Eval.ipynb

```diff
@@ -56,124 +53,187 @@
-    "os.environ[\"OPENAI_API_KEY\"] = getpass(\"Enter your OpenAI API key: \")\n",
-    "print(\"---\")\n",
-    "#####\n",
-    "\n",
-    "# 🏠 Enter your W&B project name\n",
-    "weave_client = weave.init(\"MY_PROJECT_NAME\")  # 🐝 Your W&B project name"
+    "# Setup your wandb project (team name / project name) 🐝\n",
+    "WANDB_ENTITY_PROJECT = \"my-team-name/my-project-name\"\n",
+    "\n",
+    "\n",
+    "class JsonModel(weave.Model):\n",
+    "    prompt: weave.Prompt = weave.StringPrompt(\n",
+    "        dedent(\"\"\"\n",
+    "You are an assistant that answers questions about JSON data provided by the user. The JSON data represents structured information of various kinds, and may be deeply nested. In the first user message, you will receive the JSON data under a label called 'context', and a question under a label called 'question'. Your job is to answer the question with as much accuracy and brevity as possible. Give only the answer with no preamble. You must output the answer in XML format, between <answer> and </answer> tags.\n",
+    "\"\"\")\n",
+    "    )\n",
+    "    model: str = \"OpenPipe/Qwen3-14B-Instruct\"\n",
+    "    _client: OpenAI\n",
+    "\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self._client = OpenAI(\n",
+    "            base_url=\"https://api.inference.wandb.ai/v1\",\n",
+    "            api_key=os.environ[\"WANDB_API_KEY\"],\n",
+    "            project=WANDB_ENTITY_PROJECT,\n",
+    "        )\n",
+    "\n",
+    "    @weave.op\n",
+    "    def predict(self, context: str, question: str) -> str:\n",
+    "        response = self._client.chat.completions.create(\n",
+    "            model=self.model,\n",
+    "            messages=[\n",
+    "                {\"role\": \"system\", \"content\": self.prompt.format()},\n",
+    "                {\n",
+    "                    \"role\": \"user\",\n",
+    "                    \"content\": f\"Context: {context}\\nQuestion: {question}\",\n",
+    "                },\n",
+    "            ],\n",
+    "        )\n",
+    "        return response.choices[0].message.content\n",
+    "\n",
+    "\n",
+    "@weave.op\n",
+    "def correct_answer_format(answer: str, output: str) -> dict[str, bool]:\n",
+    "    parsed_output = re.search(r\"<answer>(.*?)</answer>\", output, re.DOTALL)\n",
+    "    if parsed_output is None:\n",
+    "        return {\"correct_answer\": False, \"correct_format\": False}\n",
+    "    return {\"correct_answer\": parsed_output.group(1) == answer, \"correct_format\": True}\n",
+    "\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    if not os.environ.get(\"WANDB_API_KEY\"):\n",
+    "        print(\n",
+    "            \"WANDB_API_KEY is not set - make sure to export it in your environment or assign it in this script\"\n",
+    "        )\n",
+    "        exit(1)\n",
+    "\n",
+    "    # Find your wandb API key at: https://wandb.ai/authorize\n",
```

Comment:
I don't think we should need to do this twice in the same notebook. Every time we handle sensitive data we increase the potential security risk.